File: OOP/python/Inheritance/Multiple.py
# Multiple Inheritance Example

class AudioTrack:
    def __init__(self, bitrate):
        self.bitrate = bitrate


class VisualSequence:
    def __init__(self, resolution):
        self.resolution = resolution


class VideoClip(AudioTrack, VisualSequence):  # Multiple inheritance
    def __init__(self, bitrate, resolution, duration):
        AudioTrack.__init__(self, bitrate)
        VisualSequence.__init__(self, resolution)
        self.duration = duration

    def details(self):
        print(f"Bitrate: {self.bitrate} kbps, Resolution: {self.resolution}, Duration: {self.duration}s")


clip = VideoClip(256, "1080p", 180)
clip.details()

File: OOP/java/Inheritance/Multiple.java
package Inheritance;
// Interface A
interface AudioTrack {
    void playAudio();
}

// Interface B
interface VisualSequence {
    void playVideo();
}

// Child Class implements both
class VideoClip implements AudioTrack, VisualSequence {

    public void playAudio() {
        System.out.println("Playing audio track...");
    }

    public void playVideo() {
        System.out.println("Playing video sequence...");
    }

    void merge() {
        System.out.println("Merging audio and video...");
    }
}

public class Multiple {
    public static void main(String[] args) {
        VideoClip clip = new VideoClip();
        clip.playAudio();
        clip.playVideo();
        clip.merge();
    }
}
